src/main/java/frc/robot/RobotContainer.java

```diff
@@ -23,18 +23,50 @@ public class RobotContainer {
 	// Replace with CommandPS4Controller or CommandJoystick if needed
 	private final CommandXboxController driverController = new CommandXboxController(
 			OperatorConstants.DRIVER_CONTROLLER_PORT);
+	private final CommandXboxController XboxController = new CommandXboxController.getHID();
+	private final CommandXboxController kYButton = new XboxController.getYbutton();
```

Comment:
1. Use the XboxController object initialized in the above line, not a new class
2. `.getYButton()` returns a boolean, not a CommandXboxController object. The variable type of kYButton should be updated to match

The problem with this is that you are calling the value of the Y button ONCE, during the initialization of robotContainer, and storing it in `kYbutton`. It will NOT update automatically in this approach. Instead, you should call `xboxController.getYButton()` in a `periodic()` setting. This variable `kYButton` will only ever return the same value, because it is only set/update once